src/Main.hs

```diff
@@ -31,52 +31,48 @@ data Shape = Sphere Vector Double
             deriving (Show, Eq)
 data Light = PointLight Vector
 data Camera = Camera Double Vector Vector
-data Scene = Scene [Shape] [Light] Camera
+data Scene = Scene [Object] [Light] Camera Config
+data Config = Config { sceneWidth :: Int,
+                       sceneHeight :: Int,
+                       defaultColor :: Color
+}
 data Ray = Ray {origin :: Vector, direction :: Vector}
 
-sceneWidth :: Int
-sceneWidth = 500
-
-sceneHeight :: Int
-sceneHeight = 500
-
-spheres :: [Shape]
-spheres = [Sphere (Vector (-3) 3.5 (-8)) 3, Sphere (Vector 3 3.5 (-8)) 1]
-
-objects :: [Object]
-objects = [Object 
-                (Sphere (Vector (-3) 3.5 (-8)) 3) 
-                (Material Color.red),
-           Object
-                (Sphere (Vector 1.5 3.5 (-6)) 3)
-                (Material Color.green)]
+main :: IO()
+main =
+  let
+    objects :: [Object]
+    objects = [Object
+                    (Sphere (Vector (-3) 3.5 (-8)) 3)
+                    (Material Color.red),
+               Object
+                    (Sphere (Vector 1.5 3.5 (-6)) 3)
+                    (Material Color.green)]
 
-lights :: [Light]
-lights = [PointLight (Vector (-30) (-10) 20) ]
+    lights :: [Light]
+    lights = [PointLight (Vector (-30) (-10) 20) ]
 
-camera :: Camera
-camera = Camera 45 (Vector 0 1.8 10) (Vector 0 3 0)
+    camera :: Camera
+    camera = Camera 45 (Vector 0 1.8 10) (Vector 0 3 0)
 
-scene :: Scene
-scene = Scene spheres lights camera
+    config = Config 500 500 Color.white
```

Comment:
hah, yeah fair, though the issue with global variables is always going to be confusion with scoping. at least using them in a let block limits that.